src/openms/include/OpenMS/CHEMISTRY/ModificationsDB.h

```diff
@@ -175,7 +176,7 @@ namespace OpenMS
     std::vector<ResidueModification*> mods_;
 
     /// Stores the mappings of (unique) names to the modifications
-    Map<String, std::set<const ResidueModification*> > modification_names_;
+    std::unordered_map<String, std::set<const ResidueModification*> > modification_names_;
```

Comment:
have you tried using a plain `vector<String>` for searching (you'd need a second vector for the keys)? This might be even faster, especially if you put the most common modifications first

Reply:
you mean a sorted vector and using binary search in the vector? No I have not tried. In principle this should be O(1) and binary search is O(log(N)) so it should be slower. I think actually the way to make this faster would be to not have 10k entries in the map in the first place. We cant possibly need that many modifications.

Reply:
No binary search, but plain front-to-back search (making use of cache and prefetcher). Especially when frequently used entries (Oxidation,...) are upfront, this should always beat binary search.
Vs. HashMaps it may or may not be faster... but requires less memory in any case.

Reply:
Turns out that lookup (if you only do it once) is not that bad, maybe 10% of the total time. However, iterating overall results is bad and takes much more time around 30% of time since for each lookup of "Oxidation" we have to evaluate 13 `ResidueModification*` objects which takes time. These two things together explain most of the time spent in `ModificationsDB`